File: typescript/src/shapes/circle.ts
import { Shape } from "./shape.js";

function circle(radius: number): Shape {
    return {
        getBaseEff: () => Math.PI * radius,
        getHeightEff: () => radius,
        getArea: function (): number {
            return this.getBaseEff() * this.getHeightEff();
        }
    }
}

export { circle }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Circle.java
package edu.cmu.cs.cs214.lab02.shapes;

public class Circle implements Shape {
    private double radius;

    public Circle(double radius) {
        this.radius = radius;
    }

    public double getBaseEff() {
        return Math.PI * radius; // define the base to be pi * r
    }

    public double getHeightEff() {
        return radius; // define the height to 
    }
}
